src/App.tsx

```diff
@@ -1,29 +1,29 @@
-import { Component } from 'react';
-import './App.scss';
+import { useState } from 'react';
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
+import data from './api/movies.json';
 
-interface State {
-  movies: Movie[];
-}
+import './App.scss';
 
-export class App extends Component<{}, State> {
-  state: State = {
-    movies: [],
-  };
+export const App: React.FC = () => {
+  const [movies, setMovies] = useState(data);
 
-  render() {
-    const { movies } = this.state;
+  const addMovie = (movie: Movie) => {
+    if (movies.find(oldMovie => oldMovie.imdbId === movie.imdbId)) {
+      return;
+    }
 
-    return (
-      <div className="page">
-        <div className="page-content">
-          <MoviesList movies={movies} />
-        </div>
-        <div className="sidebar">
-          <FindMovie />
-        </div>
+    setMovies([...movies, movie]);
```

Comment:
use function to get previous movies
```suggestion
    setMovies((currentMovies) => [...currentMovies, movie]);
```